### PR TITLE
Create/edit internal user

### PR DESCRIPTION
**Pull request by sergibondarenko**
This closes
https://floragunn.atlassian.net/browse/ITT-2063

# To-do: 
- [x] UI
- [x] Fetch and store user
- [x] Hide password fields when its user update. Add a switch to show fields.
- [x] Add modal to confirm an attribute deletion. The task was moved to a separate issue https://floragunn.atlassian.net/browse/ITT-2101 

# Screenshots
Screenshot 2019-04-30 at 12 56 59
Note, there is no "update JSON" button. It is  because the JSON is updated automatically on the model  changes.
Screenshot 2019-04-30 at 12 57 17
Validation example. Form will not be submitted if the username input is empty.
Screenshot 2019-04-30 at 12 57 30

### CODE REVIEW

**Comment by sergibondarenko** [comment]
In https://github.com/floragunncom/search-guard-kibana-plugin/pull/188/commits/3275947cf5e70f504c22939416f1694ef092cb73 commit I introduced ComboBox to manage backend roles.
ComboBox

**Comment by sergibondarenko** [comment]
I noticed that having JSON editor on the right side introduces big space between elements to the form on the left. This is because of the editor height and EuiFlex stuff nature.
ugly space
I think the best way to solve it is to move JSON editor to Flyout component. Kibana does it for JSON editors.

**Comment by sergibondarenko** [comment]
In https://github.com/floragunncom/search-guard-kibana-plugin/pull/188/commits/eeb7b097ed829f029c17e5ef42cdf98760cea8ed commit I introduced Flyout.

Normal view:
Screenshot 2019-05-01 at 23 20 23
View with the Flyout:
Screenshot 2019-05-02 at 00 13 36

Notice, there is a switch "Change Password" it is visible only if you edit a user. If it is a new user, you see those password inputs.
Screenshot 2019-05-02 at 00 14 02

Plus there is an error is the passwords don't match.
Screenshot 2019-05-02 at 00 13 11

**Comment by sergibondarenko** [comment]
In https://github.com/floragunncom/search-guard-kibana-plugin/pull/188/commits/9972b2d5d396628b2be2295f2fc06797eb28c5d3 commit I added the original validation and made the password match validation look native.
Screenshot 2019-05-02 at 11 09 12

**Comment by sergibondarenko** [comment]
@mgustafsson1 As we agreed, I used the old AngularJS services, I added them in https://github.com/floragunncom/search-guard-kibana-plugin/pull/188/commits/51b556bde169578bd0d29d4c5f697a2a87cdd1a2 commit. The code looks slightly different because I had to fix multiple syntax errors.

**Comment by mgustafsson1** [comment]
> @mgustafsson1 As we agreed, I used the old AngularJS services, I added them in [51b556b](https://github.com/floragunncom/search-guard-kibana-plugin/commit/51b556bde169578bd0d29d4c5f697a2a87cdd1a2) commit. The code looks slightly different because I had to fix multiple syntax errors.

Can we do this in a way that doesn't break the git history? You never know when that comes in handy. Can't we leave the files where they were?
Also, I'm very curious about what those syntax errors were...

**Comment by mgustafsson1** [comment]
@sergibondarenko This is a bit too much to review in detail at this point, but all in all it looks good.
My main concern is the git history part, as mentioned above.
Also, I think we should reconsider using the full screen for forms and lists - there's a lot of redundant space and some buttons get aligned far away from the rest of the content etc. Let's discuss that with the team.

**Comment by sergibondarenko** [comment]
> > @mgustafsson1 As we agreed, I used the old AngularJS services, I added them in [51b556b](https://github.com/floragunncom/search-guard-kibana-plugin/commit/51b556bde169578bd0d29d4c5f697a2a87cdd1a2) commit. The code looks slightly different because I had to fix multiple syntax errors.
> 
> Can we do this in a way that doesn't break the git history? You never know when that comes in handy. Can't we leave the files where they were?

I think you are right here. It is better to keep them in the original place to preserve history. And we remove these AngularJS services completely after they are refactored to JS classes, Redux or somewhere else we find suitable. I'll adjust the code.

> Also, I'm very curious about what those syntax errors were...

Just open any file with a JS editor which has eslint configured and you see all those errors. For example, it is how I see them in atom
Screenshot 2019-05-06 at 10 41 40

**Comment by sergibondarenko** [comment]
> Also, I think we should reconsider using the full screen for forms and lists - there's a lot of redundant space and some buttons get aligned far away from the rest of the content etc. Let's discuss that with the team.

@mgustafsson1 could you prepare some example(s) for this discussion?

**Comment by sergibondarenko** [comment]
> > Also, I think we should reconsider using the full screen for forms and lists - there's a lot of redundant space and some buttons get aligned far away from the rest of the content etc. Let's discuss that with the team.
> 
> @mgustafsson1 could you prepare some example(s) for this discussion?

Also, consider that on other pages maybe we can have the forms aligned in two columns. In this case, that space to the right comes handy.

File: public/apps/configuration-react/pages/CreateInternalUser/containers/CreateInternalUser/CreateInternalUser.js [hunk 1]
```diff
@@ -62,6 +65,8 @@ class CreateInternalUser extends Component {
         let user = await internalUsersService.get(id);
         user = userToFormik({ user, id });
         this.setState({ user });
+      } else {
+        this.setState({ user: userToFormik({ user: DEFAULT_USER, id }), isEdit: !!id });
```
Comment (sergibondarenko) [inline public/apps/configuration-react/pages/CreateInternalUser/containers/CreateInternalUser/CreateInternalUser.js]: As a result,  you get a new user form after clicking on the "Create User" breadcrumb https://github.com/floragunncom/search-guard-kibana-plugin/pull/187#discussion_r281140763

**Pull request merged**